### PR TITLE
[CALCITE-6549] Add LOG1P function (enabled in Spark library)

### PR DESCRIPTION
**Pull request by caicancai**

### CODE REVIEW

**Review by NobiGo — APPROVED** [review]

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -7372,6 +7372,26 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
...
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
+    f.checkScalarApprox("log1p(0)", "DOUBLE",
+        isWithin(0.0, 0.000001));
+    f.checkScalarApprox("log1p(1)", "DOUBLE",
+        isWithin(0.6931471805599453, 0.000001));
+    f.checkScalarApprox("log1p(1e+22)", "DOUBLE",
+        isWithin(50.65687204586901, 0.000001));
+    f.checkNull("log1p(cast(null as real))");
+    f.checkNull("log1p(-1)");
+    f.checkNull("log1p(null)");
```
[the remark this answers is not in the document]
Comment (NobiGo) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:7396]: ```select log1p()```
maybe throw an exception with the invalid number of arguments to function LOG1P.

**Review by YiwenWu — APPROVED** [review]
LGTM

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (NobiGo) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:7396]: What do you mean the exception didn't show up in the test?
```
    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
    f.checkFails("^log1p()^", INVALID_ARGUMENTS_NUMBER , false);
```
I test it and looks good. What is the exception now?

[the remark this answers is not in the document]
Comment (caicancai) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:7396]: done, thank you so much

**Comment by caicancai** [comment]
@NobiGo Hi, Is it convenient to merge? https://github.com/apache/calcite/pull/3954 I plan to push this PR merge next week. I am currently testing it, and it may cause some conflicts at that time.

**Pull request merged**